src/sql-parser/src/parser.rs

```diff
@@ -2701,27 +2701,49 @@ impl<'a> Parser<'a> {
         }))
     }
 
-    /// Parse the name of a CREATE SUBSOURCE optional parameter
-    fn parse_create_subsource_option_name(
-        &mut self,
-    ) -> Result<CreateSubsourceOptionName, ParserError> {
-        let name = match self.expect_one_of_keywords(&[EXTERNAL, PROGRESS])? {
-            EXTERNAL => {
-                self.expect_keyword(REFERENCE)?;
-                CreateSubsourceOptionName::ExternalReference
-            }
-            PROGRESS => CreateSubsourceOptionName::Progress,
-            _ => unreachable!(),
-        };
-        Ok(name)
-    }
-
-    /// Parse a NAME = VALUE parameter for CREATE SUBSOURCE
     fn parse_create_subsource_option(&mut self) -> Result<CreateSubsourceOption<Raw>, ParserError> {
-        Ok(CreateSubsourceOption {
-            name: self.parse_create_subsource_option_name()?,
-            value: self.parse_optional_option_value()?,
-        })
+        let option =
+            match self.expect_one_of_keywords(&[EXTERNAL, PROGRESS, TEXT, IGNORE, DETAILS])? {
+                EXTERNAL => {
+                    self.expect_keyword(REFERENCE)?;
+                    CreateSubsourceOption {
+                        name: CreateSubsourceOptionName::ExternalReference,
+                        value: self.parse_optional_option_value()?,
+                    }
+                }
+                PROGRESS => CreateSubsourceOption {
+                    name: CreateSubsourceOptionName::Progress,
+                    value: self.parse_optional_option_value()?,
```

Comment:
Haven't reached the planner changes yet but we should update it to assert that no value was given now that this is a possibility.

Reply:
We were already using `self.parse_optional_option_value()` for this option (see a few lines above). The option is 'extracted' as a `bool` which I think allows the user to specify `PROGRESS = false` if they want to. These options are also already confirmed valid in planning:
https://github.com/MaterializeInc/materialize/blob/a20bc980786e0efa25320326449dc17481774717/src/sql/src/plan/statement/ddl.rs#L1451-L1458